main/utils.c

```diff
@@ -1322,16 +1331,29 @@ static void append_lock_information(struct ast_str **str, struct thr_lock_info *
 	int j;
 	ast_mutex_t *lock;
 	struct ast_lock_track *lt;
+	struct timeval held_for;
+	struct timeval now = ast_tvnow();
 
-	ast_str_append(str, 0, "=== ---> %sLock #%d (%s): %s %d %s %s %p (%d%s)\n",
+	held_for = ast_tvsub(now, lock_info->locks[i].last_locked);
+
+	ast_str_append(str, 0, "=== ---> %sLock #%d (%s): %s %d %s %s %p\n"
+						"===      %ld.%06ld, %ld.%06ld, %ld.%06ld (%d, %d%s)\n",
 				   lock_info->locks[i].pending > 0 ? "Waiting for " :
 				   lock_info->locks[i].pending < 0 ? "Tried and failed to get " : "", i,
 				   lock_info->locks[i].file,
 				   locktype2str(lock_info->locks[i].type),
 				   lock_info->locks[i].line_num,
-				   lock_info->locks[i].func, lock_info->locks[i].lock_name,
+				   lock_info->locks[i].func,
+				   lock_info->locks[i].lock_name,
 				   lock_info->locks[i].lock_addr,
+				   lock_info->locks[i].last_locked.tv_sec,
```

Comment:
This output is already a bit unfriendly at times, for the time based stuff can it be made more user friendly in output?

Reply:
maybe cut it to deltas?

Reply:
Whatever we actually think is useful, of which I'm probably not a good person to say because I rarely use this.